stats/stats.go

```diff
@@ -203,24 +206,47 @@ func NewHistogram(Offset float64, Divider float64) *Histogram {
 	return h
 }
 
-// Tradeoff memory for speed (though that also kills the cache so...)
-// this creates an array of 100k (max value) entries
-// TODO: consider using an interval search for the last N big buckets
+// Val2Bucket values are kept in two different structure
+// val2Bucket allows you reach between 0 and 1000 in constant time
 func init() {
-	lastV := int32(lastValue)
-	val2Bucket = make([]int, lastV+1)
+	val2Bucket = make([]int, maxArrayValue)
+	maxArrayValueIndex = -1
+	for i, value := range histogramBucketValues {
+		if value == maxArrayValue {
+			maxArrayValueIndex = i
+			break
+		}
+	}
+	if maxArrayValueIndex == -1 {
+		log.Fatalf("Bug boundary maxArrayValue=%d not found in bucket list %v", maxArrayValue, histogramBucketValues)
+	}
 	idx := 0
-	for i := int32(0); i <= lastV; i++ {
+	for i := int32(0); i < maxArrayValue; i++ {
 		if i >= histogramBucketValues[idx] {
 			idx++
 		}
 		val2Bucket[i] = idx
 	}
-	// coding bug detection (aka impossible if it works once)
-	if idx != numValues {
-		log.Fatalf("Bug in creating histogram buckets idx %d vs numbuckets %d (last val %d)", idx, numValues, lastV)
+	// coding bug detection (aka impossible if it works once) until 1000
+	if idx != maxArrayValueIndex {
+		log.Fatalf("Bug in creating histogram index idx %d vs index %d up to %d", idx, int(maxArrayValue), maxArrayValue)
 	}
+}
 
+// lookUpIdx looks for scaledValue's index in histogramBucketValues
+// TODO: change linear time to O(log(N)) with binary search
+func lookUpIdx(scaledValue int) int {
+	scaledValue32 := int32(scaledValue)
+	if scaledValue32 < maxArrayValue { //constant
+		return val2Bucket[scaledValue]
+	}
+	for i := maxArrayValueIndex; i < numValues; i++ {
+		if histogramBucketValues[i] > scaledValue32 {
+			return i
+		}
+	}
+	//select max value
+	return numValues - 1
```

Comment:
does this statement get triggered? the last bucket isn't numValues - 1
I'd prefer something clearly off if it's not supposed to happen (this time the argument about maxValue already checked in record() should apply)

Reply:
This case will be used when scaledValue is equals to 100000. because in the for loop,  the highest check is 100000 > scaledValue. IF value is bigger than 100000, it will be handled by record method.